flink-staging/flink-streaming/flink-streaming-connectors/src/main/java/org/apache/flink/streaming/connectors/kafka/api/KafkaSink.java

```diff
@@ -114,17 +132,24 @@ public KafkaSink(String zookeeperAddress, String topicId,
 	public void open(Configuration configuration) {
 
 		KafkaTopicUtils kafkaTopicUtils = new KafkaTopicUtils(zookeeperAddress);
-		String brokerAddress = kafkaTopicUtils.getLeaderBrokerAddressForTopic(topicId);
+		String listOfBrokers = kafkaTopicUtils.getBrokerList(topicId);
+
+		if (LOG.isInfoEnabled()) {
+			LOG.info("Broker list: {}", listOfBrokers);
+		}
 
-		props = new Properties();
+		if (props == null) {
```

Comment:
How about always setting our default properties (broker list, max retries etc.) and overwrite them with the properties passed by the users.